enhancements/sig-policy/62-namespace-labelselector/README.md

```diff
@@ -0,0 +1,199 @@
+The `LabelSelector` struct includes `MatchLabels` and `MatchExpressions`. In addition to aligning
+with Kubernetes practices, this will cut down on processing the namespace list for clusters with a
+large number of namespaces since a shorter list would be returned from the Kubernetes API.
+
+Futhermore, an additional proposal is to use the `filepath` Go library rather than a custom wildcard
+implementation for the `include` and `exclude` lists. This will bring maintainability and additional
+power to the namespace filtering. This will not be a breaking change since the current behavior is
+to only support leading and trailing wildcards, and a pattern like `foo*bar` is treated as a literal
+string, which is not a valid namespace name.
+
+### Goals
+
+1. Filter the namespace list by `LabelSelector` for the `NamespaceSelector`
+2. Allow `LabelSelector` and `include`/`exclude` arrays to operate together when both are provided
+   or separately when provided individually
+3. Replace custom wildcard code with `filepath`
+
+## Proposal
+
+Implement `MatchLabels` and `MatchExpressions` alongside the `include` and `exclude` arrays:
+
+```yaml
+namespaceSelector:
+  include: []
+  exclude: []
+  matchLabels: {}
+  matchExpressions: []
+```
+
+These can easily be used to construct a `LabelSelector` for the Kubernetes API to consume.
+
+The `MatchLabels` and `MatchExpressions` are AND-ed by default, so the implementation will carry
+this forward to find the intersection of the `LabelSelector` and, if provided, the `include` and
+`exclude` arrays. Either the `include` and `exclude` arrays or the `LabelSelector` can be provided.
+If the `LabelSelector` is not provided, all namespaces are retrieved and filtered with the `include`
+and `exclude` arrays.
+
+### User Stories
+
+Suppose clusters have namespaces named `test1` through `test5` labeled with a `name` key and value
+matching its name and an `all: namespaces` label.
+
+#### Story 1
+
+As an administrator of `ConfigurationPolicies`, I can choose namespaces in which to deploy defined
+objects by the namespace label.
+
+```yaml
+namespaceSelector:
+  matchLabels:
+    name: test2
+  matchExpressions:
+    key: name
+    operator: In
+    values: ["test1", "test2"]
+```
+
+**Result**: `test2`
+
+#### Story 2
+
+As an administrator of `ConfigurationPolicies`, I can use `include`/`exclude` filepath patterns to
+filter out by namespace name, and without needing to provide `MatchLabels`/`MatchExpressions`.
+
+```yaml
+namespaceSelector:
+  include: ["test[3-5]"]
+  exclude: ["test4"]
+```
+
+**Result**: `test3`, `test5`
+
+#### Story 3
+
+As an administrator of `ConfigurationPolicies`, I can choose namespaces in which to deploy defined
+objects by the namespace label and then filter out by name using filepath patterns.
+
+```yaml
+namespaceSelector:
+  exclude: ["test[2-3]"]
+  matchLabels:
+    all: namespaces
+```
+
+**Result**: `test2`, `test3`
```

Comment:
I think you mean `include` here, or the result should be `test1`, `test4`, `test5`? Or I'm missing something

Reply:
Good catch! Here I was thinking I was being tricky...and apparently I was. 😅